homeassistant/components/axis/media_player.py

```diff
@@ -0,0 +1,146 @@
+from homeassistant.helpers.httpx_client import get_async_client
+
+from . import AxisConfigEntry
+from .entity import AxisEntity
+from .hub import AxisHub
+
+_LOGGER = logging.getLogger(__name__)
+
+
+async def async_setup_entry(
+    hass: HomeAssistant,
+    config_entry: AxisConfigEntry,
+    async_add_entities: AddEntitiesCallback,
+) -> None:
+    """Add media player for an Axis Network Speaker."""
+    hub = config_entry.runtime_data
+    async_add_entities([AxisSpeaker(hub)])
+
+
+class AxisSpeaker(AxisEntity, MediaPlayerEntity):
+    """Axis Network Speaker."""
+
+    _attr_has_entity_name = True
+    _attr_name = None
+    _attr_device_class = MediaPlayerDeviceClass.SPEAKER
+    _attr_media_content_type = MediaType.MUSIC
+    _attr_supported_features = (
+        MediaPlayerEntityFeature.PLAY_MEDIA
+        | MediaPlayerEntityFeature.VOLUME_SET
+        | MediaPlayerEntityFeature.VOLUME_STEP
+        | MediaPlayerEntityFeature.STOP
+        | MediaPlayerEntityFeature.BROWSE_MEDIA
+    )
+    entity_description: MediaPlayerEntityDescription
+
+    def __init__(self, hub: AxisHub) -> None:
+        """Initialize the entity."""
+        super().__init__(hub)
+        self._attr_unique_id = f"{hub.unique_id}_mediaplayer"
+        self._attr_state = MediaPlayerState.IDLE
+
+    async def async_browse_media(
+        self,
+        media_content_type: MediaType | str | None = None,
+        media_content_id: str | None = None,
+    ) -> BrowseMedia:
+        """Implement the websocket media browsing helper."""
+        return await media_source.async_browse_media(
+            self.hass,
+            media_content_id,
+            content_filter=lambda item: item.media_content_type.startswith("audio/"),
+        )
+
+    async def async_play_media(
+        self, media_type: MediaType | str, media_id: str, **kwargs: Any
+    ) -> None:
+        """Play a piece of media."""
+        if media_source.is_media_source_id(media_id):
+            media_type = MediaType.MUSIC
+            play_item = await media_source.async_resolve_media(
+                self.hass, media_id, self.entity_id
+            )
+            media_id = async_process_play_media_url(self.hass, play_item.url)
+
+        if media_type != MediaType.MUSIC:
+            raise HomeAssistantError("Only music media type is supported")
+
+        _LOGGER.debug("playing media %s for axis speaker", media_id)
+
+        ffmpeg_manager = ffmpeg.get_ffmpeg_manager(self.hass)
+        output = await ffmpeg_cmd(ffmpeg_manager.binary, media_id)
+
+        uri = "/axis-cgi/audio/transmit.cgi"
+        conf = self.hub.config
+        url = f"{conf.protocol}://{conf.host}:{conf.port}{uri}"
+        auth = httpx.DigestAuth(self.hub.config.username, self.hub.config.password)
+        headers = {"Content-Type": "audio/axis-mulaw-128"}
+
+        client = get_async_client(self.hass, verify_ssl=False)
+        await client.post(url, headers=headers, content=output, auth=auth)
```

Comment:
Use the `self.hub.api.vapix.request` method, you should probably move everything related to the request into the axis library, keep ffmpeg stuff here
```suggestion
        headers = {"Content-Type": "audio/axis-mulaw-128"}
        await self.hub.api.vapix.request("post", uri, headers=headers, content=output)
```